pallets/maintenance-mode/src/lib.rs

```diff
@@ -181,6 +181,24 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> Pallet<T> {
+		/// Internal function to force the chain into maintenance mode without an origin check.
+		pub fn force_enter_maintenance_mode() {
```

Comment:
Also, lets call this method from the `enter_maintenance_mode` extrinsic, removing duplicated code.